src/ocean/meta/types/ReduceType.d

```diff
@@ -0,0 +1,199 @@
+/*******************************************************************************
+
+    Generic utility allowing to recursively visit an arbitrary type and reduce
+    its definition to some compile-time value. It is intended to be used as
+    implementation cornerstone for complex type traits to address recursive
+    type reflection boilerplate being copied multiple times.
+
+    Copyright:
+        Copyright (c) 2017 Sociomantic Labs GmbH.
+        All rights reserved.
+
+    License:
+        Boost Software License Version 1.0. See LICENSE_BOOST.txt for details.
+        Alternatively, this file may be distributed under the terms of the Tango
+        3-Clause BSD License (see LICENSE_BSD.txt for details).
+
+*******************************************************************************/
+
+module ocean.meta.types.ReduceType;
+
+import ocean.meta.traits.Basic;
+
+/*******************************************************************************
+
+    Params:
+        T = arbitrary type to visit/reduce
+        Reducer = struct type conforming `assertValidReducer` requirements used
+            to do actual calculations/checks on the type
+
+    Example:
+
+    Example reducer that counts amount of integer typed entities accessible
+    though the base types (i.e. fields and referenced types):
+
+    ---
+    static struct ExampleReducer
```

Comment:
There is already unittest with same code, I copied it here for the purpose of doc rendering only.